### DISCUSSION TITLE
Can oal data use visual TopList to present data

### DISCUSSION THREAD

**Discussion by AllenArbiter** [apache/skywalking#11037] [General]
I used oal to collect data and TopList to show data failed
  - name: aws_connector_execute_time
    exp: aws_connector_execute_time.avg(['service', 'name']) The visual component Line can be displayed normally ，But with TopList, you can't，Is there a use example

**Comment by wu-sheng** [comment]
Could you explain a little about `TopList`? I don't remember if we supported that. cc @kezhenxu94

**Reply by AllenArbiter** [reply]
<img width="1262" alt="image" src="https://github.com/apache/skywalking/assets/37923716/c6e1bca0-50ad-4d57-9c89-a2f3dbb7d3c6">

**Reply by AllenArbiter** [reply]
I was able to display my data using Line, but not using TopList So I wonder if it's because my data collection came from oal

**Reply by wu-sheng** [reply]
But that is time series data. How a list helps?

**Comment by AllenArbiter** [comment]
I don't know how to collect the data of topList and show it. I think it is the same as I collected the data of line before, but I can't find relevant documents. Could you please give me an example It is best to use oal collection and then use topList display

**Reply by wu-sheng** [reply]
TopList is to ordering the value.

**Reply by wu-sheng** [reply]
@Fine0830 Is this asking a sorted metric or something?

**Comment by Fine0830** [comment]
Here is an example for  the `topList`.

`{
    "data": {
        "expression0": {
            "type": "SORTED_LIST",
            "results": [
                {
                    "metric": {
                        "labels": []
                    },
                    "values": [
                        {
                            "name": "19f4d9836c914f9b8dfd9d0c994681e2@10.116.1.213",
                            "value": "100",
                            "refId": null
                        },
                        {
                            "name": "1d7d178cce68440ea69e5a8b5f513947@10.116.1.221",
                            "value": "100",
                            "refId": null
                        },
                        {
                            "name": "1f4523e4e39c4f5692a1464b81ebe585@10.116.1.198",
                            "value": "100",
                            "refId": null
                        },
                        {
                            "name": "ed8033374d8646bdb4b30066ac81d529@10.116.5.30",
                            "value": "100",
                            "refId": null
                        }
                    ]
                }
            ],
            "error": null
        }
    }
}`

**Reply by wu-sheng** [reply]
OK, for sorted list, I don't think it works for labeled value. That is value nothing shows up. @AllenArbiter 

The single value metric could be sorted.

**Reply by AllenArbiter** [reply]
Thank you very much for your reply. It seems to be because of mine labeled value Caused by
I used to collect line data in code like this
MeterFactory.counter(SlaPullMetric.AWSPaaS_MVCFramework_topThread)
                    .tag("name", fastTrackThread.getThread().getName())
                    .mode(CounterMode.INCREMENT)
                    .build()
                    .increment(100);
Now how can I collect the data on the java side in order to collect the sample format data you gave，It would be better if there was documentation for me to learn on my own, or if I could see if I could be told how to collect the data in the format you provide

**Reply by wu-sheng** [reply]
Just a thing that, if you have labeled, it can't be sorted. You need to aggregate them through MAL. 
Nothing special. 

What @Fine0830 provided is query format, which is for me.

**Reply by AllenArbiter** [reply]
I'm still a little confused，This is the way I use it now
```
MeterFactory
			.counter("sql-count")
			.mode(CounterMode.RATE)
			.build()
                         .increment(100)
```
I used the MeterFactory counter to collect data and then went to MAL
```
  - name: sql-count
    exp: sql-count.sum(['service'])
```
I can show it in TopList, but if we don't let tag, how can we bring up the corresponding number of sql, or what kind of API can be used to collect such data? This data is very consistent with our situation, we can change the value of name to SQL to achieve a query SQL corresponding return data amount，Then use TopN to show it

{
    "data": {
        "expression0": {
            "type": "SORTED_LIST",
            "results": [
                {
                    "metric": {
                        "labels": []
                    },
                    "values": [
                        {
                            "name": "select *from orguser",
                            "value": "100",
                            "refId": null
                        },
                        {
                            "name": "select * from wfh_task",
                            "value": "3111",
                            "refId": null
                        },
                        {
                            "name": "select * from cdk_io",
                            "value": "211",
                            "refId": null
                        },
                        {
                            "name": "select * from wfh_task_detail",
                            "value": "211",
                            "refId": null
                        }
                    ]
                }
            ],
            "error": null
        }
    }
}

**Reply by wu-sheng** [reply]
There is no such case on our side for this. Slow SQLs are usually sampled through logs(such as MySQL Server monitoring), and only ordered, but not aggregated. But you are trying to run statistics based on SQL statements?

**Reply by AllenArbiter** [reply]
Yes,sir. We have a requirement to display query SQL return quantity monitoring，I can now intercept my client server's query SQL code，He then reports the number of queries he returns to Skywalking, I used Line before and everything worked fine，But our boss found that you have a very slow SQL monitoring works very well，We want to do a [query data volume SQL sort] to monitor the data

This makes me completely confused, I don't know how to collect the above JSON data

**Reply by wu-sheng** [reply]
If you are confused, how can I understand better than you do? We have already provided both ways, a) sampled slow SQL, 2) MAL to run any statistics on any labels(in your case, maybe it is SQL statement oriented).
I can't see what I can do more to help.
